### DISCUSSION TITLE
My approach to allow widget to debounce refresh

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#6654] [General]
https://github.com/tiddly-gittly/flowtiwi-sidebar/blob/3c8c28a1e1428dfabd346fa71a6b77a21291cf8b/src/flowtiwi-sidebar.ts#L13-L28

```ts
  constructor(parseTreeNode: any, options: any) {
    super(parseTreeNode, options);
    const debouncedRefresh = debounce(this.refresh.bind(this), DEBOUNCE_UPDATE_INTERVAL);
    this.refresh = (changedTiddlers: IChangedTiddlers) => {
      debouncedRefresh(changedTiddlers);
      return true;
    };
  }

  refresh(changedTiddlers: IChangedTiddlers) {
    return this.refreshChildren(changedTiddlers);
  }
```

In this way, there won't be frequent update on this expensive widget. But the drawback is my `refresh` will always return `true` now. Because denounced function don't have a return value, I have to return `true` every time.

**Comment by Jermolene** [comment]
Hi @linonetwo I don't think that this is strictly safe. The core expects the entire refresh cycle to run to completion synchronously before any other asynchronous tasks can run. The code here causes some of the refresh processing to be deferred out of synchronous execution, which breaks all the expectations of object lifecycle.

**Reply by linonetwo** [reply]
People is having 10k+ tiddlers https://talk.tiddlywiki.org/t/how-to-disable-all-lists-update-on-every-click-to-make-tiddlywiki-work-fast-with-10000-tiddlers/2313/23

Mine has 15k+ tiddlers. So when widgets like Echarts refresh on each editor typing, it feels not very good. And the solutions might be 

1. auto caching in the core, for frequently called filter expressions, and debounce the stale-while-revalidating
2. use sqlite to speedup core API
3. debounce the widget
